source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -48,6 +50,92 @@ bool cbsContainsU16(CBS& cbs, uint16_t n) {
   return false;
 }
 
+int sslTlsContextIndex() {
+  CONSTRUCT_ON_FIRST_USE(int, []() -> int {
+    int ssl_context_index = SSL_CTX_get_ex_new_index(0, nullptr, nullptr, nullptr, nullptr);
+    RELEASE_ASSERT(ssl_context_index >= 0, "");
+    return ssl_context_index;
+  }());
+}
+
+bool seekToSubject(CRYPTO_BUFFER* cert, CBS* tbs_certificate) {
```

Comment:
Can this (and the other similar functions) return optional<CBS>? I'm assuming CBS is just a pointer and length (not a big struct).

Reply:
It's a little bit more than that:
https://boringssl.googlesource.com/boringssl/+/74161f485b5d54fe963cbd3d081b718ec84d2e00/include/openssl/bytestring.h#39
let me know how do you want to proceed.